importers/link_16_importer.py

```diff
@@ -0,0 +1,199 @@
+from pepys_import.file.importer import CANCEL_IMPORT, Importer
+from pepys_import.utils.sqlalchemy_utils import get_lowest_privacy
+from pepys_import.utils.text_formatting_utils import (
+    custom_print_formatted_text,
+    format_error_message,
+)
+from pepys_import.utils.unit_utils import convert_absolute_angle, convert_distance, convert_speed
+
+V1_HEADER = "PPLI"
+V2_HEADER = "Xmt/Rcv"
+
+
+class Link16Importer(Importer):
+    """Importer to handle two different formats of track information that are
+    transmitted using Link-16 encoding
+    """
+
+    def __init__(self):
+        super().__init__(
+            name="Link-16 Format Importer",
+            validation_level=constants.BASIC_LEVEL,
+            short_name="Link-16 Importer",
+            default_privacy="Private",
+            datafile_type="Link-16",
+        )
+        self.version = 1
+
+    def can_load_this_type(self, suffix):
+        return suffix.upper() == ".CSV"
+
+    def can_load_this_filename(self, filename):
+        return True
+
+    def can_load_this_header(self, header):
+        # V1 starts w/ PPLI
+        # V2 starts w/ Xmt/Rcv
+        return header.startswith(V1_HEADER) or header.startswith(V2_HEADER)
+
+    def can_load_this_file(self, file_contents):
+        return True
+
+    def _load_this_file(self, data_store, path, file_object, datafile, change_id):
+        # Read the base timestamp from the filename
+        filename, ext = os.path.splitext(os.path.basename(path))
+        datetime_string = self.extract_timestamp(filename)
+        self.base_timestamp = self.timestamp_to_datetime(datetime_string)
+        self.current_hour = 0
+        self.previous_time = timedelta(hours=0, minutes=0, seconds=0)
+
+        # Now do what we'd normally do on load
+        for line_number, line in enumerate(tqdm(file_object.lines()), 1):
+            result = self._load_this_line(data_store, line_number, line, datafile, change_id)
+            if result == CANCEL_IMPORT:
+                custom_print_formatted_text(
+                    format_error_message(f"Error in file caused cancellation of import of {path}")
+                )
+                break
+
+    def _load_this_line(self, data_store, line_number, line, datafile, change_id):
+        if line_number == 1:
+            # Will only parse if one of these headers found
+            self.version = 1 if line.text.startswith(V1_HEADER) else 2
+            return
+        tokens = line.tokens(line.CSV_TOKENISER, ",")
+        if len(tokens) <= 1:
+            # To skip over any blank lines
+            return
+
+        # Time wrangling - TODO - consider extracting into a method
+        time_token = tokens[1]
+        line_time = timedelta(
+            hours=0,
+            minutes=int(time_token.text[:2]),
+            seconds=float(time_token.text[3:]),
+        )
+
+        if line_time < self.previous_time:
+            self.current_hour += 1
+
```

Comment:
@mew-nsc - I was going to ask for a Slack call to for you to talk me through the logic in lines 90 & 91, but an explanation in comments would be a more persistent way of capturing what's happening. Would you mind explaining what's happening?